src/gpuhunt/providers/vultr.py

```diff
@@ -0,0 +1,181 @@
+import logging
+from typing import Any, Optional
+
+import requests
+from requests import Response
+
+from gpuhunt import QueryFilter, RawCatalogItem
+from gpuhunt._internal.constraints import KNOWN_AMD_GPUS, KNOWN_NVIDIA_GPUS
+from gpuhunt._internal.models import AcceleratorVendor
+from gpuhunt.providers import AbstractProvider
+
+logger = logging.getLogger(__name__)
+
+API_URL = "https://api.vultr.com/v2"
+
+EXCLUSION_LIST = ["GH200"]
+
+
+class VultrProvider(AbstractProvider):
+    NAME = "vultr"
+
+    def get(
+        self, query_filter: Optional[QueryFilter] = None, balance_resources: bool = True
+    ) -> list[RawCatalogItem]:
+        offers = fetch_offers()
+        return sorted(offers, key=lambda i: i.price)
+
+
+def fetch_offers() -> Optional[list[RawCatalogItem]]:
+    """Fetch plans with types:
+    1. Cloud GPU (vcg),
+    2. Bare Metal (vbm),
+    3. and other CPU plans, including:
+        Cloud Compute (vc2),
+        High Frequency Compute (vhf),
+        High Performance (vhp),
+        All optimized Cloud Types (voc)"""
+    bare_metal_plans_response = _make_request("GET", "/plans-metal?per_page=500")
+    other_plans_response = _make_request("GET", "/plans?type=all&per_page=500")
+    return convert_response_to_raw_catalog_items(bare_metal_plans_response, other_plans_response)
+
+
+def convert_response_to_raw_catalog_items(
+    bare_metal_plans_response: Response, other_plans_response: Response
+) -> list[RawCatalogItem]:
+    catalog_items = []
+
+    bare_metal_plans = bare_metal_plans_response.json()["plans_metal"]
+    other_plans = other_plans_response.json()["plans"]
+
+    for plan in bare_metal_plans:
+        for location in plan["locations"]:
+            catalog_item = get_bare_metal_plans(plan, location)
+            if catalog_item:
+                catalog_items.append(catalog_item)
+
+    for plan in other_plans:
+        for location in plan["locations"]:
+            catalog_item = get_instance_plans(plan, location)
+            if catalog_item:
+                catalog_items.append(catalog_item)
+
+    return catalog_items
+
+
+def get_bare_metal_plans(plan: dict, location: str) -> Optional[RawCatalogItem]:
+    gpu_name, gpu_count = extract_gpu_info_from_id(plan["id"])
+    if gpu_name in EXCLUSION_LIST:
+        logger.info(f"Excluding plan with GPU {gpu_name} as it is not supported.")
+        return None
+    gpu_memory = get_gpu_memory(gpu_name) if gpu_name else None
+    gpu_vendor = get_gpu_vendor(gpu_name)
+    return RawCatalogItem(
+        instance_name=plan["id"],
+        location=location,
+        price=round(plan["monthly_cost"] / 730, 2),
+        cpu=plan["cpu_count"],
```

Comment:
Judging by the [pricing page](https://www.vultr.com/pricing/#bare-metal), bare metal instances have true CPUs rather than vCPUs. True CPUs have twice as many threads, which are roughly equivalent to vCPUs.

![image](https://github.com/user-attachments/assets/a1062bfa-0c5d-452b-8600-ef5619a40a5d)

Could you please verify it by connecting to a bare metal instance and checking the number of CPUs, e.g. by running `htop`? It should appear as if the instance has twice as many CPUs. If that's the case, we should multiply `plan["cpu_count"]` by two for bare metal. We already do it [for Oracle](https://github.com/dstackai/gpuhunt/blob/b707e743b2c27d7ec9be17e87e689997cbcbbeaf/src/gpuhunt/providers/oci.py#L250).

Note that this doesn't apply to ARM CPUs. At the moment the only ARM CPU in Vultr seems to be the GH200, but they may add more in the future. Since there doesn't seem to be a way to tell whether the CPU is ARM by looking at Vultr's API, I'd also suggest whitelisting all bare metal instances we currently support, so that ARM instances don't appear in the catalog with incorrect `cpu` values once (if) Vultr adds them.